### ISSUE TITLE
Strange TV Static like sound.

### ISSUE THREAD

**Issue by YiffLord** [thewierdnut/endless-mobile#127]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Upon entering the game there is a constant yet subtle static like sound.

### Steps to Reproduce

Boot up the game.

### Expected Behavior

No static? 

### Device type

Overclocked Samsung Tab S2 9.7 Black

### Game Version

Latest Continuous as of 25/2/2024

### Additional Information

I removed every plugin I had in case they were creating that weird static, that did not remove it.

**Comment by thewierdnut** [comment]
I'm not seeing this. Can I get a screen capture?

**Comment by YiffLord** [comment]
> I'm not seeing this. Can I get a screen capture?

Excuse the late reply, couldn't get online sooner...

I'll try to get it in a moment.

Additional information: This static is not affected by the sound volume bar.

**Comment by YiffLord** [comment]
https://github.com/thewierdnut/endless-mobile/assets/89931675/6ab77de2-29a0-4529-b846-e7bc7cc85207

Now, I must say the screen recorder makes it _slightly_ worse than it is. It is however noticeable.
It's not a predominant sound on the game, it is rather a background overlap static. It isn't affected by any volume modifier wether it is the game's or the device's.

**Comment by thewierdnut** [comment]
That sounds to me like it is repeatedly playing a short ~10ms buffer over and over again. Something may have broken in the openAL layer. 

Do any of the game's normal sound effects still work?

**Comment by thewierdnut** [comment]
Also... what happens if you background the app, then pull it into the foreground again? Does the noise continue in the background? Does it stay the same when you foreground the app?

**Comment by YiffLord** [comment]
> Also... what happens if you background the app, then pull it into the foreground again? Does the noise continue in the background? Does it stay the same when you foreground the app?

If I exit the app the noise stops, that is until I enter back in. Like most non background apps do with sounds within them. I.e. Music & SFX, when you background them they stop playing in an instant, then when you foreground them the sound goes back as it was.

**Comment by YiffLord** [comment]
> That sounds to me like it is repeatedly playing a short ~10ms buffer over and over again. Something may have broken in the openAL layer.
> 
> Do any of the game's normal sound effects still work?

Every SFX in the game works properly as far as I know. This static does not interfere with game sounds, it is just a constant overlap. If in-game Sfx get loud enough (shooting or thrusting for example) this static is completely obscured by them and goes unnoticed, yet it never stops.

**Comment by thewierdnut** [comment]
It sounds like a looping sound effect is turned on, and then left on without ever getting shut off. I'll see if I can instrument a build to visually show the active sound effects or something.

**Comment by YiffLord** [comment]
Alright sounds like you have a pretty solid idea of what is, that's great.

Do you have any other questions before I go offline?

**Comment by YiffLord** [comment]
Alright sounds like you have a pretty solid idea of what is, that's great.

**Comment by thewierdnut** [comment]
> Do you have any other questions before I go offline?

No, and I won't get to an instrumented build for a few days either.

**Comment by YiffLord** [comment]
Alright, good luck & have a nice day/night.

**Comment by thewierdnut** [comment]
The latest continuous build should now be instrumented to print all sounds to the screen if the `Show CPU / GPU load` preference is turned on.

**Comment by YiffLord** [comment]
> The latest continuous build should now be instrumented to print all sounds to the screen if the `Show CPU / GPU load` preference is turned on.

I'm gonna try it out.

**Comment by YiffLord** [comment]
![Screenshot_20240304-140335](https://github.com/thewierdnut/endless-mobile/assets/89931675/158e5046-3567-4a0e-8664-5abfe10ac3cb)
![Screenshot_20240304-140349](https://github.com/thewierdnut/endless-mobile/assets/89931675/31f9de25-72e0-4d6d-ac96-1e73f0af6b42)


The static persists.

**Comment by YiffLord** [comment]
It is most noticeable in the main menu, however the MM does not have any sound print

**Comment by YiffLord** [comment]
Oddly enough the static goes away when there is a sound playing.

To notice that I put the volume bar to the minimum possible without entirely removing sounds, as previously mentioned the static is consistently loud regardless of any sound meter. The sound effects print, and they're dim enough to not obscure the static. Once a sound effect plays, the static goes off, then reappears when no sound plays.

**Comment by thewierdnut** [comment]
I'm at a loss at this point. If the sound isn't being printed on the screen there, it isn't one that endless-sky is trying to play, which means something is broken between openAL and openSL. 

I haven't seen any changes in my code or upstream that change the audio. Is this issue present in builds 38, 39, 41 or 42?

**Comment by YiffLord** [comment]
I only use the Continuous, not any specific builds but I am confident the issue persists.

I did have hints of the static in previous versions, down to 10.3 iirc but I've only fully noticed since I started using "high end" headphones.

If I had the PC version I could check if the same happened in the base game...
Mayhaps the adaptation to mobile just fucks with it a different way, I don't know.

**Comment by thewierdnut** [comment]
I can't reproduce this with any version, with desktop or mobile builds. I'm not sure what else to try :(

**Comment by YiffLord** [comment]
Could a signature modification cause it in any shape, way or form?

**Comment by thewierdnut** [comment]
no

**Comment by YiffLord** [comment]
> no

Oh well then I'm at a complete loss aswell...

**Comment by YiffLord** [comment]
Guess I just gotta deal with it, I don't mind really.

**Comment by DefaultS3ttings** [comment]
Are you charging your tablet whilst playing ES? My phones produce that static noise when using a particular charger.

**Comment by YiffLord** [comment]
> Are you charging your tablet whilst playing ES? My phones produce that static noise when using a particular charger.

No, not the cause...
